src/app/store/product-detail/product-detail.component.ts

```diff
@@ -0,0 +1,32 @@
+import { Component, OnInit, Input } from '@angular/core';
+import { ActivatedRoute, Params } from '@angular/router';
+import { Product } from 'app/core/store/product.model';
+import { StoreService } from 'app/core/store/store.service';
+
+@Component({
+  selector: 'app-product-detail',
+  templateUrl: './product-detail.component.html',
+  styleUrls: ['./product-detail.component.scss']
+})
+export class ProductDetailComponent implements OnInit {
+  @Input() product: Product = new Product();
+
+  constructor(private route: ActivatedRoute,
+    private storeService: StoreService) { }
+
+  ngOnInit() {
+    this.storeService.getProduct(this.route.snapshot.params['id']).subscribe(result => {
+      this.product = result;
+    })
+
+    this.route.params
+    .subscribe(
+      (params: Params) => {
+        this.storeService.getProduct(params['id']).subscribe(result => {
+          this.product = result;
+        })
+      }
```

Comment:
This looks like duplicate logic on line 18 and then on 22. Like possibly lines 18-20 were written, but then it didn't work, and then lines 22 - 28 were written to subscribe to the param change. So, one of these can be deleted right?

And then if you need to keep lines 22-18, like promises, you don't want to nest observables, these should be chained rather than nested. This is pseudo code but something like

```
this.route.params
  .flatMap((params: Params) => this.storeService.getProduct(params.id))
  .subscribe(result => this.product = result);
```
That should be more or less correct. Notice i never have a subscribe inside another subscribe callback. Once you do that, you are effectively back in "callback hell", where logic and error handling gets tricky. Promises and observables were made partly to avoid having to do that.